docs/router/src/example/navigation-targets.md

```diff
@@ -0,0 +1,157 @@
+# Navigation Targets
+In the previous chapter we learned how to create links to pages within our app.
+We told them where to go using the `target` property. This property takes a
+[`NavigationTarget`].
+
+## What is a navigation target?
+A [`NavigationTarget`] is similar to the `href` of an HTML anchor element.It
+tells the router where to navigate to. The Dioxus Router knows three kinds of
+navigation targets:
+- [`NtPath`]: we already saw that. It's basically an `href`, but cannot link to
+  content outside our app.
+- [`NtExternal`]: This works exactly like an HTML anchors `href`. In fact, it is
+  just passed through. Don't use this for in-app navigation as it'll trigger a
+  page reload by the browser.
+- [`NtName`]: this is the most interesting form of navigation target. We'll look
+  at it in detail in this chapter.
+
+## External navigation
+If we need a link to an external page we can do it like this:
+```rust,no_run
+# // Hidden lines (like this one) make the documentation tests work.
+# extern crate dioxus;
+# use dioxus::prelude::*;
+#
+#[allow(non_snake_case)]
+fn GoToDioxus(cx: Scope) -> Element {
+    cx.render(rsx! {
+        Link {
+            target: NtExternal(String::from("https://dioxuslabs.com")),
+            "Explicit NtExternal target"
+        }
+        Link {
+            target: "https://dioxuslabs.com".into(),
+            "Implicit NtExternal target"
+        }
+    })
+}
+```
+
+> Note that we can use a `str` with `.into()`, just like with [`NtPath`]
+> targets. The router will convert a `str` to a [`NtExternal`] only if it starts
+> with either `http://` or `https://`.
+
+## Named navigation
+When defining our routes, we can optionally give them unique static names. This
+is required for a feature we call named navigation.
+
+Up to now, when creating links we told the router the exact path to go to. With
+named navigation we instead give it a name, and let it figure out the path.
```

Comment:
Oh, very nice! Definitely better than `str`s . However, users could still provide invalid names, e.g. by forgetting to add them to their route.

Reply:
This works now! Great idea! I was thinking about recommending `const`s, but this is much, much better.

Reply:
I implemented an API that works almost like your example. Besides `QNone` becoming `None`, there are two differences I'll list below. Also, I'm considering a third change and would like to get a second opinion.

### `.name` instead of `.identifier`
I kept the name _name_, because from an app dev perspective, you still give the route a name. The only difference is that the name is not stored in the value you provide, but in its type.

Also, the same thing is referred to as _name_ in Vue and Laravel. (If React Router has this feature, I can't find it.)

### `NamedTarget` (the successor of `NtName`)
`NamedTarget` now is a bit more complex. It needs to keep track of the `TypeId` and a `&str` of the type name. The former is needed for identifying the route. The later is necessary to show the name in error messages, otherwise devs would only see an unhelpful `u64`.

To keep `NamedTarget` as easy to use as it was before, it can now be constructed via `Into` like this:
```rust
rsx! { Link {
    target: (SomeName, vec![], None),
    "Some Link"
} }
```

### Using type parameters directly
I'm thinking about changing the API to use type parameters directly. This could look something like this:
```rust
// defining a named route
Route::new(Something as Component).name::<SomeName>()

// creating a NamedTarget
named_target::<SomeName>(vec![], None)
```

I think the disadvantage is obvious, as the API is a bit more complicated to use. However, it'd be much clearer that the value of the name doesn't matter. Consider this example with the current implementation:
```rust
Route::new(Something as Component).name("first name");
Route::new(Something as Component).name("other name");
```

As you can see, both lines look quite different in code. However, both share the same name type and as such would conflict with each other.

Reply:
We will tweak this to be a bit more ergonomic.